pkg/api/v1alpha1/cluster.go

```diff
@@ -674,6 +682,14 @@ func validateMirrorConfig(clusterConfig *Cluster) error {
 	if clusterConfig.Spec.RegistryMirrorConfiguration.InsecureSkipVerify && clusterConfig.Spec.DatacenterRef.Kind != SnowDatacenterKind {
 		return errors.New("insecureSkipVerify is only supported for snow provider")
 	}
+
+	if clusterConfig.Spec.RegistryMirrorConfiguration.Authenticate {
+		username := os.Getenv("REGISTRY_USERNAME")
+		password := os.Getenv("REGISTRY_PASSWORD")
+		if username == "" || password == "" {
+			return fmt.Errorf("username or password not set, Provide REGISTRY_USERNAME and REGISTRY_PASSWORD to use authenticated registry mirror")
+		}
+	}
```

Comment:
Ideally we should separate reading configuration from validating it. This enables changing the source of configuration without creating a bunch of touch points. 

On a side note, we're duplicating environment variables across the code base as `REGISTRY_USERNAME` and `REGISTRY_PASSWORD` are specified elsewhere also. Are we confident their semantics are the same in all places? If so, we should be using constants rather than re-specifying.